WoodStar/WoodStar/WoodStarConnection.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Buffers;
+using System.IO;
+using System.Net;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WoodStar
+{
+    public class WoodStarConnection
+    {
+        private readonly string _serverIp;
+        private readonly string _username;
+        private readonly string _password;
+        private readonly string _database;
+
+        private TdsChannel? _tdsChannel;
+
+        public WoodStarConnection(string serverIp, string username, string password, string database)
+        {
+            _serverIp = serverIp;
+            _username = username;
+            _password = password;
+            _database = database;
+        }
+
+        public async Task OpenAsync()
+        {
+            var host = Dns.GetHostEntry(_serverIp);
+            var ipAddress = host.AddressList[1];
+            var remoteEP = new IPEndPoint(ipAddress, 1433);
+            var socket = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+            await socket.ConnectAsync(remoteEP);
+            var networkStream = new NetworkStream(socket);
+            _tdsChannel = new TdsChannel(networkStream);
+            await _tdsChannel.Prelogin();
+
+            var interceptingStream = new InterceptingStream(networkStream);
+            var sslStream = new SslStream(interceptingStream, leaveInnerStreamOpen: false);
+            await sslStream.AuthenticateAsClientAsync(new SslClientAuthenticationOptions()
+            {
+                TargetHost = _serverIp,
+                RemoteCertificateValidationCallback = (_, _, _, _) => true,
+            });
+            interceptingStream.StopIntercepting();
+            _tdsChannel = new TdsChannel(sslStream);
+            await _tdsChannel.Login();
```

Comment:
One final problem - while a login request has to be send over the `sslStream`, the response has to be read over a non-ssl stream.